src/lib/erc20.ts

```diff
@@ -0,0 +1,30 @@
+import { ThirdwebContract, toTokens } from "thirdweb";
+import { getActiveClaimCondition } from "thirdweb/extensions/erc20";
+import { getContractMetadata } from "thirdweb/extensions/common";
+import { fetchCurrencyMetadata } from "@/lib/utils";
+
+export async function getERC20Info(contract: ThirdwebContract) {
+  const [claimCondition, contractMetadata] = await Promise.all([
+    getActiveClaimCondition({ contract }).catch((err) => {

```

Comment:
What's the point of catch just to throw?

Just don't catch anything in here at all!

I want to see only ONE try catch in the the main top level Home component and THATS IT